### PR TITLE
Bug 2116715: remove GenericResourceInformer code because it is not used anymore

### PR DESCRIPTION
**Pull request by atiratree**

### CODE REVIEW

File: pkg/cmd/controller/route/interfaces.go [hunk 1]
```diff
@@ -54,8 +52,7 @@ type ControllerContext struct {
 	RouteInformers    routeinformer.SharedInformerFactory
 	OperatorInformers operatorinformer.SharedInformerFactory
 
-	GenericResourceInformer genericinformers.GenericResourceInformer
```
Comment (deads2k) [inline pkg/cmd/controller/route/interfaces.go:57]: Just delete this instance and everything in this file using it.  The other non-route controllers need it.

Comment (atiratree) [reply pkg/cmd/controller/route/interfaces.go:57]: seems like it moved all away, to the cluster-policy-controller

**Comment by ingvagabund** [comment]
```
FAIL: github.com/openshift/openshift-controller-manager/pkg/serviceaccounts/controllers TestUpdateOldStyleSecretWithKey 45.01s
17
make: *** [vendor/github.com/openshift/build-machinery-go/make/targets/golang/test-unit.mk:12: test-unit] Error 1
18
{"component":"entrypoint","error":"wrapped process failed: exit status 2","file":"k8s.io/test-infra/prow/entrypoint/run.go:80","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2022-08-08T17:56:59Z"} 
```

Checking locally:
```
$ go test -v github.com/openshift/openshift-controller-manager/pkg/serviceaccounts/controllers -run TestUpdateOldStyleSecretWithKey
=== RUN   TestUpdateOldStyleSecretWithKey
    docker_registry_service_test.go:274: Waiting for ready
I0808 20:26:03.546700 2529014 docker_registry_service.go:145] Starting DockerRegistryServiceController controller
    docker_registry_service_test.go:281: Waiting to reach syncRegistryLocationHandler
    docker_registry_service_test.go:287: Waiting to update secret
--- PASS: TestUpdateOldStyleSecretWithKey (0.10s)
PASS
I0808 20:26:03.647575 2529014 docker_registry_service.go:163] Shutting down DockerRegistryServiceController controller
ok  	github.com/openshift/openshift-controller-manager/pkg/serviceaccounts/controllers	0.132s
```

/retest

**Review by soltysh — APPROVED** [review]
/approve

**Pull request merged**